server/registry/names/common.go

```diff
@@ -43,7 +43,9 @@ func GenerateID() string {
 
 // validateID returns an error if the provided ID is invalid.
 func validateID(id string) error {
-	if !customIdentifier.MatchString(id) {
+	if "" == id {
+		return fmt.Errorf("invalid identifier %q: identifier is required", id)
```

Comment:
I think you chose this wording for the CLI (makes sense) but this could also be triggered if someone sets the id to "". Would "nonempty identifier is required" be clear enough to cover all cases?

Reply:
I chose it to match all the other error messages below it.

Reply:
Right, makes sense. So how about 
"invalid identifier %q: nonempty identifier is required" 
or
"invalid identifier %q: identifier must be nonempty"
? I slightly prefer the first (particularly for the CLI case).

Reply:
sure